credential-provider/src/services/core.service.ts

```diff
@@ -36,6 +38,87 @@ export class CoreService {
     return res;
   }
 
+  /**
+   * Performs the following to create response token for interaction-hook
+   *
+   * 1. Get secret from interaction-hook config - MattrService.getOpenIdConfig() -> res.data.secret
+   * 2. Get tenant_domain from ENV/config -> issuer
+   * 3. Get URL for interaction-hook UI -> NGROK_URL/core/2fa -> audience
+   * 4. Validate session_token using info above -> verifyResult
+   * 5. Extract state & redirectUrl from verifyResult.payload
+   * 6. Create responseTokenPayload using state & any claims
+   * 7. Create responseToken using SignJWT(...etc)
+   * 8. Construct callbackUrl for redirect on interaction-hook
+   * 9 Return callbackUrl
+   * @param CreateResponseTokenArgs
+   * @returns string
+   * @example
+   * await createResponseToken({
+   *   session_token: "ONE_TIME_JWT_TOKEN",
+   * })
+   */
+  public async createResponseToken(
+    args: CreateResponseTokenArgs,
+  ): Promise<string> {
+    const { session_token } = args;
+
+    const getOpenIdConfigRes = await this.mattrService.getOpenIdConfig({
+      token: this.config.get('MATTR_AUTH_TOKEN'),
+    });
+    const encodedSecret = getOpenIdConfigRes.data.interactionHook.secret;
+    this.logger.log(`res.data.interactionHook.secret  -> ${encodedSecret}`);
+    const secret = Buffer.from(encodedSecret, 'base64');
+    this.logger.log(`secret -> ${JSON.stringify(secret)}`);
+
+    const decoded = decodeJwt(session_token);
+    const audience = decoded.aud as string;
+    const issuer = decoded.iss;
+    this.logger.log(`issuer > ${issuer}`);
+    const verifyResult = await jwtVerify(session_token, secret, {
+      issuer,
+      audience,
+    }).catch((error) => {
+      this.logger.error('Invalid session token', error);
+      throw new Error('Invalid session token');
+    });
+    this.logger.log('Verified session token', verifyResult);
+
+    const { state, redirectUrl } = verifyResult.payload;
+
+    const responseTokenPayload = {
+      /**
+       * IMPORTANT: The state must be signed to prevent CSRF attacks.
+       */
+      state,
+      /**
+       * The claims to be merged is optional.
+       */
+      claims: {},
+    };
+    const responseToken = await new SignJWT(responseTokenPayload)
+      .setProtectedHeader({ alg: 'HS256', typ: 'JWT' })
+      .setIssuedAt()
+      .setExpirationTime('1m')
+      .setIssuer(audience as string)
```

Comment:
It's already being casted to string above? Ideally we should handle the different types